linode/domainzonefile/framework_datasource.go

```diff
@@ -0,0 +1,136 @@
+package domainzonefile
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	meta *helper.FrameworkProviderMeta
```

Comment:
why don't we use `client *linodego.Client` here as we normally do?

Reply:
The client is already defined in the `helper.FrameworkProviderMeta` so I removed it to reduce redundancy.